geth/peers/peerpool_test.go

```diff
@@ -84,20 +84,41 @@ func (s *PeerPoolSimulationSuite) TestSingleTopicDiscovery() {
 	defer subscription.Unsubscribe()
 	s.NoError(peerPool.Start(s.peers[1]))
 	defer peerPool.Stop()
-	connected := 0
-	for {
-		select {
-		case ev := <-events:
-			if ev.Type == p2p.PeerEventTypeAdd {
-				connected++
-			}
-		case <-time.After(5 * time.Second):
-			s.Require().FailNowf("waiting for peers timed out", strconv.Itoa(connected))
+	var connected discover.NodeID
+	select {
+	case ev := <-events:
+		if ev.Type == p2p.PeerEventTypeAdd {
+			connected = ev.Peer
 		}
-		if connected == expectedConnections {
-			break
+	case <-time.After(5 * time.Second):
+	}
+	s.Equal(s.peers[0].Self().ID, connected)
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Why do we need timers in unit tests?

Reply:
stopping discovery is done in separate goroutine, so i need either this or add some kind of event feed with channels. do we want one just for tests?

Reply:
I always prefer a kind of management of goroutines so that I can stop them as well as waiting until they are stopped. But got my little helper for it, see https://godoc.org/github.com/tideland/gotogether.

Reply:
it is not about stopping a goroutine but stopping a discovery server in another goroutine. that goroutine is kinda a manager of discovery and discovery server.